components/link/link.js

```diff
@@ -88,7 +99,12 @@ class Link extends FocusMixin(LitElement) {
 			 * Where to display the linked URL
 			 * @type {string}
 			 */
-			target: { type: String }
+			target: { type: String },
+			/**
+			 * Whether to display the open in new window icon
+			 * @type {boolean}
+			 */
+			newWindow: { type: Boolean, attribute: 'new-window' }
```

Comment:
What are folks' thoughts about NOT making the opt-in? Like essentially just showing the indicator on all `<d2l-link>`s whenever `target="_blank">`. A quick code search only yields ~30 places using links with blank targets. We could do a quick audit to confirm that they'd all want this functionality.

It would dramatically simplify this code, but also the main benefit would be immediate UX/a11y gains for users.

Reply:
Even if we don't show the icon, forcing the hidden is span whern target="_blank" is an immediate accessibility benefit

Reply:
Here's a list of all the places this would affect:

Definitely require action:
- [Insights Assessment Quality Dashboard](https://github.com/Brightspace/insights-assessment-quality-dashboard/blob/46bbb294bfc726e1bcd7e96a763416430aa145ad/pages/quiz-item-analysis-view.js#L481): has a "go to quiz" link and this is actually already rendering the "new window" icon. We'd need to ensure this got cleaned up to avoid 2 icons. It ALSO has an `aria-label` which has text indicating it opens in a new window, so that would need to be removed.
- [Wave](https://github.com/Brightspace/nova/blob/92a0fdb3f944fbc2092eca4ffc95ee10549223d9/src/app/main/components/applications/application-overview.js#L311): is also rendering the "new window" icon itself

We should confirm with teams that they're OK with this:
- [Parent Portal 1](https://github.com/Brightspace/parent-portal-ui/blob/c0ea044a9ece75c883f20afdb49bb89e3b7d2152/src/pages/no-children.js#L67): support text with "click here" link.
- [Parent Portal 2](https://github.com/Brightspace/parent-portal-ui/blob/c0ea044a9ece75c883f20afdb49bb89e3b7d2152/src/components/parent-teacher-interview-alert.js#L82): parent-teacher interview link
- [User Profile Card](https://github.com/BrightspaceHypermediaComponents/users/blob/6fedf39419f95be319b39590110215af217342d7/components/user-profile-card/user-profile-card.js#L505): for showing a user's website link
- [Import Quiz Questions 1](https://github.com/Brightspace/import-quiz-questions/blob/bb7e9c6cb788dc4dd0846117f21cdb389b781ec5/src/components/FileUpload/SupportedFormatsInfo.jsx#L10): link to download a CSV template. There's also a test doing this that'll need updating.
- [Import Quiz Questions 2](https://github.com/Brightspace/import-quiz-questions/blob/bb7e9c6cb788dc4dd0846117f21cdb389b781ec5/src/components/ErrorPage/ErrorPage.jsx#L64): wouldn't be impacted directly as it's using JavaScript to open the new window. However, it's manually rendering the "new window" icon so might benefit from being able to force it ON even if the target isn't set.
- [IPSIS Admin UI](https://github.com/Brightspace/ipsis-admin-ui/blob/3ffe3d024bc9657bc9a3889e1b37acbf8f0cc295/src/pages/dashboard-page.js#L302): "learn more" link to the Community -- probably the reason we're doing this?
- [Custom My Courses in Other Orgs](https://github.com/Brightspace/custom-my-courses-in-other-orgs/blob/b1f63178c6b6102caeaa68147ce2c1ea6a3cb044/custom-my-courses-in-other-orgs.js#L201): opens a link to an API endpoint
- [LCS Instructor Check-in Widget](https://github.com/Brightspace/lcs-aau-instructor-check-in-widget/blob/2f198cac337391ede2cf79fc85952c1c1386a568/src/shared/custom_widgets/instructor_check_in_widget/components/lcs-instructor-check-in-widget.js#L546): rendering a "get help" link and also a help icon
- [LCS core widgets](https://github.com/Brightspace/lcs-core-widgets/blob/079b08a208c8f7474765319e21be6eb70c16c430/code_le/src/lcs-cdn/widgets/profiles/X.X.X/components/admin-edit-student-view-profile.js#L500): link to student profile -- also rendering its own "opens in new window" offscreen text but no icon

Internal tools so we don't care:
- [AWS Jumpbox](https://github.com/Brightspace/aws-jumpbox/blob/e75417e29c12867a3e0cd2619d00d79f018525b0/web/src/index.html#L105)
- [Tenant Finder](https://github.com/Brightspace/tenant-finder.d2l.dev/blob/586774efe5c523a7e949b9a72a85c5bffd6fef7d/src/components/tenant-finder.js#L149)
- [CDS Catalogue](https://github.com/Brightspace/cds-catalogue/blob/f1a092632d4c444e2b74fc0fc4f5a6cc5ed476c3/src/components/cds-catalogue-modal.js#L38): innovation project
- [Instance Metadata Store](https://github.com/Brightspace/instance-metadata-store/blob/315e90ffcc0e3a05e885fc5c843e13ab671da059/ui/public/index.html#L40)
- [LCS webdev template](https://github.com/Brightspace/lcs-webdev-template/blob/f187d7968e9617637864ff384d3bd2571683ff78/src/components/d2l-component-collection.js#L170)
- [CDS Template UI Demo](https://github.com/Brightspace/cds-template-ui-demo/blob/4cd3c41a7cdc014453a45bba8322dc5a8b343b17/components/test-component.js#L40)
- [Test Reporting UI](https://github.com/Brightspace/test-reporting.d2l.dev/blob/ee3e92dbe1bdc2d9d5d2fc09efdb88544c290506/ui/src/components/test-reporting-github-actions-link.js#L32)